draft-thoji-scone-trone-protocol.md

```diff
@@ -305,17 +336,72 @@ packet and replace an existing rate signal.
 
 ~~~ pseudocode
 is_long = packet[0] & 0x80 == 0x80
-is_trone = compare(packet[1..5], TRONE_VERSION)
-if is_long and is_trone:
-  dcid_len = packet[5]
-  offset = 6 + dcid_len
-  scid_len = packet[offset]
-  offset = offset + 1 + scid_len
-  packet_rate = read_rate_signal(packet[offset : offset + rate_signal_size])
-  if target_rate.is_lower_than(packet_rate):
-    write_rate_signal(packet[offset : offset + rate_signal_size], target_rate)
+packet_version = packet[1..5]
+if is_long and (packet_version == TRONE1_VERSION or packet_version == TRONE2_VERSION):
+  target_rate_value = convert_rate_to_signal(target_rate, packet_version)
+  packet_rate_value = packet[0] & 0x3f
+  if packet_rate_value == 0x3f or target_rate_value < packet_rate_value:
+    packet[0] = packet[0] & 0xc0 | target_rate_value
 ~~~
 
+For TRONE packets with two versions, a network element applies the same rate
+policy to both, accounting for the different scales of each version. When a
+network element signals a rate limit that falls within the range of both
+versions, it should signal appropriate values in each version. When the target
+rate limit is outside the range of a particular version, the network element
+should signal the minimum value (0) for rates below the range and preserve the
+original value for rates above the range.
+
+# Version Interaction {#version-interaction}
+
+The TRONE protocol defines two versions (0xTRONE1 and 0xTRONE2) that cover
+different but overlapping ranges of bitrates. This design allows for:
+
+1. Support for both very low bitrates (down to 100 Kbps) and very high bitrates
+   (up to 12.5 Gbps)
+2. Graceful handling of network elements that might only recognize one version
+   or some subset of future versions.
```

Comment:
Would be "up to 10Gbps" if we use only 62 actual ranges.

Reply:
`10M*10^{62/20}` = 12.589G; `10M*10^{61/20}` = 11.220G.  We only get 10G if we decide to use 61 ranges.

Reply:
Sorry, I didn't do the math. But yes, that's the crux of the debate. We have 64 code points, from 0 to 63. If we reserve one to say "don't know", that means using only 62 code points, the last one marking the unbounded range.

Reply:
We can come back to that after merging this PR.